File: apis/AnnouncementsAPI/Requests/PutAnnouncementRequest.cs
using DatabaseContextLibrary.models;

namespace AnnouncementsAPI.Requests
{
    public class PutAnnouncementRequest
    {
        public string? Title { get; set; }
        public string? Description { get; set; }
        public Guid? PetId { get; set; }
        public Status? Status { get; set; }
    }
}

File: client/lib/services/announcements/requests/put_announcement_request.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:pet_share/announcements/models/announcement.dart';
import 'package:pet_share/utils/announcementstatus_converter.dart';

part 'put_announcement_request.g.dart';

@JsonSerializable()
class PutAnnouncementRequest {
  PutAnnouncementRequest({
    this.title = '',
    this.description = '',
    this.petId,
    this.status,
  });

  String? petId;
  String? title;
  String? description;
  @AnnouncementStatusConverter()
  AnnouncementStatus? status;

  Map<String, dynamic> toJson() => _$PutAnnouncementRequestToJson(this);
  factory PutAnnouncementRequest.fromJson(Map<String, dynamic> json) =>
      _$PutAnnouncementRequestFromJson(json);
}
